### PR TITLE
feat: split code in different files

### PR DESCRIPTION
**Pull request by grumbach**
- split code in different files
- rename Name trait to Proposition

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
-// #[cfg(feature = "bad_crypto")]
-// pub mod bad_crypto;
+#[cfg(feature = "bad_crypto")]
+pub mod bad_crypto;
 // #[cfg(feature = "blsttc")]
 // pub mod blsttc;
-// #[cfg(feature = "ed25519")]
-// pub mod ed25519;
+#[cfg(feature = "ed25519")]
+pub mod ed25519;
```
Comment (davidrusu) [inline src/lib.rs:10]: Keep these commented out, they don't work with the current code base

Comment (grumbach) [reply src/lib.rs:10]: It was to appease clippy because they are used for error types, should I comment them back?

Comment (davidrusu) [reply src/lib.rs:10]: Oh, sure keep them in then. I wonder why my clippy is not complaining

**Review by davidrusu — APPROVED** [review]
LGTM

**Pull request merged**